### PR TITLE
tests: minor fixes to lxd-container test bugs

### PR DESCRIPTION
**Pull request by dheyay**
## Why is this needed?

Fixes minor bugs in the lxd-container tests

## Test Steps

---

- [ ] *(un)check this to re-run the checklist action*

### CODE REVIEW

File: features/steps/ubuntu_advantage_tools.py [hunk 1]
```diff
@@ -182,7 +182,7 @@ def when_i_create_local_ppas(context, release, next_release):
     from features.steps.machines import given_a_machine
 
     # We need Kinetic or greater to support zstd when creating the PPAs
-    given_a_machine(context, "mantic", "lxd-container", machine_name="ppa")
+    given_a_machine(context, "jammy", "lxd-container", machine_name="ppa")
```
Comment (orndorffgrant) [inline features/steps/ubuntu_advantage_tools.py:185]: The comment above specifies that this needs to be Kinetic or later. Was `mantic` not working here? Does `jammy` work here?

File: features/ubuntu_upgrade.feature [hunk 2]
```diff
@@ -50,8 +50,6 @@ Feature: Upgrade between releases when uaclient is attached
       | bionic  | lxd-container | focal        | lts    |               | esm-infra | enabled         | esm-apps | enabled         | true           |
       | bionic  | lxd-container | focal        | lts    |               | usg       | enabled         | usg      | enabled         | pro enable cis |
       | focal   | lxd-container | jammy        | lts    |               | esm-infra | enabled         | esm-apps | enabled         | true           |
-      | jammy   | lxd-container | mantic       | normal |               | esm-infra | n/a             | esm-apps | n/a             | true           |
```
Comment (orndorffgrant) [inline features/ubuntu_upgrade.feature:53]: We need a test that tests an upgrade from jammy. If it now upgrades to `noble`, then we need to update this line, not remove it

File: features/ubuntu_upgrade_unattached.feature [hunk 3]
```diff
@@ -61,8 +61,6 @@ Feature: Upgrade between releases when uaclient is unattached
       | xenial  | lxd-container | bionic       | lts    |               | enabled        |
       | bionic  | lxd-container | focal        | lts    |               | enabled        |
       | focal   | lxd-container | jammy        | lts    |               | enabled        |
-      | jammy   | lxd-container | mantic       | normal |               | n/a            |
```
Comment (orndorffgrant) [inline features/ubuntu_upgrade_unattached.feature:64]: same here

File: features/steps/ubuntu_advantage_tools.py (hunk 1 above)
Comment (dheyay) [reply features/steps/ubuntu_advantage_tools.py:185]: `mantic` fails here which resulted in the upgrade errors but `jammy` works here. I recently ran it with 'noble' and that worked too so will upgrade to noble.

File: features/ubuntu_upgrade.feature (hunk 2 above)
Comment (dheyay) [reply features/ubuntu_upgrade.feature:53]: I did run the upgrade from `jammy` to `noble` but it failed and there was a comment right below which comments out the jammy -> noble upgrade.
 
 ```
# No path from Jammy to Noble until .1 is there
# | jammy   | lxd-container | noble        | lts    |                 | esm-infra | enabled         | esm-apps | enabled         | true           |
 ```

Comment (orndorffgrant) [reply features/ubuntu_upgrade.feature:53]: Sorry - now I remember. This is okay to delete

We should probably have the jammy->noble line uncommented but with the `--devel-release` flag - I'll add a separate comment with that suggestion.

File: features/steps/ubuntu_advantage_tools.py (hunk 1 above)
Comment (orndorffgrant) [reply features/steps/ubuntu_advantage_tools.py]: Sorry - that makes sense about mantic. I'm surprised jammy worked, but yes noble is the better option here. Thank you for updating it!

File: features/ubuntu_upgrade.feature [hunk 4]
```diff
@@ -50,8 +50,6 @@ Feature: Upgrade between releases when uaclient is attached
...
-      | mantic  | lxd-container | noble        | normal |               | esm-infra | disabled        | esm-apps | disabled        | true           |
 
   # No path from Jammy to Noble until .1 is there
   # | jammy   | lxd-container | noble        | lts    |                 | esm-infra | enabled         | esm-apps | enabled         | true           |
```
Comment (orndorffgrant) [inline features/ubuntu_upgrade.feature:55]:
```suggestion
  # | jammy   | lxd-container | noble        | lts    |--devel-release| esm-infra | enabled         | esm-apps | enabled         | true           |
```

File: features/steps/ubuntu_advantage_tools.py [hunk 5]
```diff
@@ -123,6 +123,17 @@ def when_i_install_uat(context, machine_name=SUT):
             )
 
 
+@when("I ensure -proposed is not enabled anymore")
+def when_i_ensure_proposed_not_enabled(context, machine_name=SUT):
+    if context.pro_config.install_from is InstallationSource.PROPOSED:
+        when_i_run_command(
```
Comment (renanrodrigo) [inline features/steps/ubuntu_advantage_tools.py:129]: I think either this should read `I disable the proposed pocket`, or it should run `apt update` right after - to actually ensure no proposed data is in the lists cache.

File: features/ubuntu_upgrade.feature (hunk 2 above)
Comment (renanrodrigo) [reply features/ubuntu_upgrade.feature:53]: devel release won't work anymore, because devel is now oracular. Commented is good.

Comment (orndorffgrant) [reply features/ubuntu_upgrade.feature:53]: LTS' are weird prior to next LTS.1
- `Prompt=lts` says "There is no development version of an LTS available."
- `Prompt=normal` + `--devel-release` looks at oracular and says "Upgrades to the development release are only available from the latest supported release."
- `Prompt=lts` + `--devel-release` upgrades successfully no noble :upside_down_face:

Comment (renanrodrigo) [reply features/ubuntu_upgrade.feature:53]: alright good good

**Review by orndorffgrant — APPROVED** [review]

**Pull request merged**